core/src/main/java/org/elasticsearch/common/logging/ESLoggerFactory.java

```diff
@@ -19,62 +19,29 @@
 
 package org.elasticsearch.common.logging;
 
-import org.elasticsearch.common.logging.jdk.JdkESLoggerFactory;
-import org.elasticsearch.common.logging.log4j.Log4jESLoggerFactory;
-import org.elasticsearch.common.logging.slf4j.Slf4jESLoggerFactory;
-import org.elasticsearch.common.settings.AbstractScopedSettings;
+import org.apache.log4j.Logger;
 import org.elasticsearch.common.settings.Setting;
-import org.elasticsearch.common.settings.Settings;
 
 import java.util.Locale;
-import java.util.Map;
-import java.util.function.Consumer;
-import java.util.regex.Pattern;
 
 /**
  * Factory to get {@link ESLogger}s
  */
 public abstract class ESLoggerFactory {
 
-    public static final Setting<LogLevel> LOG_DEFAULT_LEVEL_SETTING = new Setting<>("logger.level", LogLevel.INFO.name(), LogLevel::parse, false, Setting.Scope.CLUSTER);
-    public static final Setting<LogLevel> LOG_LEVEL_SETTING = Setting.dynamicKeySetting("logger.", LogLevel.INFO.name(), LogLevel::parse, true, Setting.Scope.CLUSTER);
-
-    private static volatile ESLoggerFactory defaultFactory = new JdkESLoggerFactory();
-
-    static {
-        try {
-            Class<?> loggerClazz = Class.forName("org.apache.log4j.Logger");
-            // below will throw a NoSuchMethod failure with using slf4j log4j bridge
-            loggerClazz.getMethod("setLevel", Class.forName("org.apache.log4j.Level"));
-            defaultFactory = new Log4jESLoggerFactory();
-        } catch (Throwable e) {
-            // no log4j
-            try {
-                Class.forName("org.slf4j.Logger");
-                defaultFactory = new Slf4jESLoggerFactory();
-            } catch (Throwable e1) {
-                // no slf4j
-            }
-        }
-    }
-
-    /**
-     * Changes the default factory.
-     */
-    public static void setDefaultFactory(ESLoggerFactory defaultFactory) {
-        if (defaultFactory == null) {
-            throw new NullPointerException("defaultFactory");
-        }
-        ESLoggerFactory.defaultFactory = defaultFactory;
-    }
-
+    public static final Setting<LogLevel> LOG_DEFAULT_LEVEL_SETTING =
+            new Setting<>("logger.level", LogLevel.INFO.name(), LogLevel::parse, false, Setting.Scope.CLUSTER);
+    public static final Setting<LogLevel> LOG_LEVEL_SETTING =
+            Setting.dynamicKeySetting("logger.", LogLevel.INFO.name(), LogLevel::parse, true, Setting.Scope.CLUSTER);
 
     public static ESLogger getLogger(String prefix, String name) {
-        return defaultFactory.newInstance(prefix == null ? null : prefix.intern(), name.intern());
+        prefix = prefix == null ? null : prefix.intern();
```

Comment:
Probably! I don't know why it was there and it was undocumented. I'm inclined to leave it in this PR and open an issue for it so we don't forget it entirely.